types/didyoumean/index.d.ts

```diff
@@ -1,6 +1,6 @@
 export = didYouMean;
 
-declare function didYouMean(str: string, list: string[], key?: string): string | string[];
+declare function didYouMean(str: string, list: string[] | object[], key?: string): string | string[];
```

Comment:
```suggestion
declare function didYouMean(str: string, list: string[] | Record<string, string>[], key?: string): string | null;
```

We could use the [Record utility type](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) to narrow it down. Also, it appears that the return value is either `string` or `null` (can be changed via `nullResultValue`). Please confirm before applying.

Reply:
Thanks for your review
I test this
```js
import didYouMean from "didyoumean";

let input = "insargrm";
let result;
const list: string[] = ["facebook", "twitter", "instagram", "linkedin"];

result = didYouMean(input, list);
console.log(result) // "instagram"

input = "google plus";

result = didYouMean(input, list);
console.log(result) // null


input = "insargrm";
const listOfObjects = [{ id: "facebook" }, { id: "twitter" }, { id: "instagram" }, { id: "linkedin" }];
const key = "id";
result = didYouMean(input, listOfObjects, key);
console.log(result) // "instagram"

didYouMean.returnWinningObject = true;
result = didYouMean(input, listOfObjects, key);
console.log(result) // "instagram"
```

The output: 
`instagram`
`null`
`instagram`
`{ id: 'instagram' }`

so I think we should add object to return value

Reply:
Sorry, missed this in the inbox. So, the proposed definition above is based on the default options. If we're considering all the customization possibilities, we'll need to specify the return value as `any` since it's achievable via the `nullResultValue` property which can be any value.

Appreciate thoughts from the maintainers in this regard.